compiler/luci/pass/src/RemoveUnnecessaryStridedSlicePass.test.cpp

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (c) 2021 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "luci/Pass/RemoveUnnecessaryStridedSlicePass.h"
+
+#include <luci/IR/CircleNodes.h>
+
+#include <gtest/gtest.h>
+
+namespace
+{
+
+void create_remove_unnecessary_strided_slice(loco::Graph *g,
+                                             const std::initializer_list<uint32_t> input_shape,
+                                             bool remove)
+{
+  assert(g);
+
+  // Input create
+  auto input = g->nodes()->create<luci::CircleInput>();
+  auto graph_input = g->inputs()->create();
+  input->index(graph_input->index());
+  input->shape_status(luci::ShapeStatus::VALID);
+  input->rank(input_shape.size());
+  input->shape(input_shape);
+
+  // Begin create
+  auto begin = g->nodes()->create<luci::CircleConst>();
+  begin->dtype(loco::DataType::S32);
+  begin->size<loco::DataType::S32>(input_shape.size());
+  begin->rank(1);
+  begin->dim(0).set(input_shape.size());
+  for (int i = 0; i < input_shape.size(); ++i)
+  {
+    begin->at<loco::DataType::S32>(i) = remove ? 0 : 1;
+  }
+
+  // Strides create
+  auto strides = g->nodes()->create<luci::CircleConst>();
+  strides->dtype(loco::DataType::S32);
+  strides->size<loco::DataType::S32>(input_shape.size());
+  strides->rank(1);
+  strides->dim(0).set(input_shape.size());
+  for (int i = 0; i < input_shape.size(); ++i)
+  {
+    strides->at<loco::DataType::S32>(i) = -1;
+  }
+
+  auto end = g->nodes()->create<luci::CircleConst>();
+  end->dtype(loco::DataType::S32);
+  end->size<loco::DataType::S32>(input_shape.size());
+  end->rank(1);
+  end->dim(0).set(input_shape.size());
+  end->shape(input_shape);
```

Comment:
```suggestion
  end->dtype(loco::DataType::S32);
  end->rank(1);
  end->dim(0).set(input_shape.size());
  end->size<loco::DataType::S32>(input_shape.size());
  for (int i = 0; i < input_shape.size(); ++i)
  {
    strides->at<loco::DataType::S32>(i) = ...;
  }
```
we don't need to set `end->shape(input_shape);` as `rank()` and `dim(..).set(..)` will set the shape.
you need to call ` strides->at<loco::DataType::S32>(i) = ...;` to set values of this const

Reply:
Thanks a lot. Fixed.